core/dataset_operations.cpp

```diff
@@ -133,7 +143,8 @@ Dataset histogram(const Dataset &dataset, const Dim &dim) {
 
 Dataset merge(const DatasetConstProxy &a, const DatasetConstProxy &b) {
   return Dataset(union_(a, b), union_(a.coords(), b.coords()),
```

Comment:
Put a comment that explains that `union_` is used deliberately, instead of `union_or`.